src/utils/genetic_operators.cpp

```diff
@@ -49,6 +49,23 @@ namespace pagmo
 namespace detail
 {
 
+// Helper function for implementation of the binary crossover.
+static inline double sbx_betaq(double beta, double eta_c, detail::random_engine_type &random_engine)
+{
+    // Random distributions
+    std::uniform_real_distribution<> drng(0., 1.); // to generate a number in [0, 1)
+    
+    double alpha = 2. - std::pow(beta, -(eta_c + 1.));
+    double rand01 = drng(random_engine);
```

Comment:
here you will be using two different drng for the two calls. In the original deb code, intead, the same is used.
maybe pass rand01 as a paramter and remove tthe rng inside here?